daemon/stores/partial.py

```diff
@@ -75,32 +77,41 @@ def add(self, args: Namespace, **kwargs) -> PartialStoreItem:
 
 
 class PartialPodStore(PartialPeaStore):
-    """A Pod store spawned inside mini-jinad container"""
+    """A Pod store spawned inside partial-daemon container"""
 
     peapod_cls = Pod
 
 
 class PartialFlowStore(PartialStore):
-    """A Flow store spawned inside mini-jinad container"""
+    """A Flow store spawned inside partial-daemon container"""
 
-    def add(self, args: Namespace, port_expose: int, **kwargs) -> PartialStoreItem:
-        """Starts a Flow in `mini-jinad`.
+    def add(self, args: Namespace, **kwargs) -> PartialStoreItem:
+        """Starts a Flow in `partial-daemon`.
 
         :param args: namespace args for the flow
-        :param port_expose: port expose for the Flow
         :param kwargs: keyword args
         :return: Item describing the Flow object
         """
         try:
             if not args.uses:
-                raise ValueError('Uses yaml file was not specified in flow definition')
+                raise ValueError('uses yaml file was not specified in flow definition')
 
             with open(args.uses) as yaml_file:
-                y_spec = yaml_file.read()
-            flow = Flow.load_config(y_spec)
-            flow.workspace_id = jinad_args.workspace_id
-            flow.port_expose = port_expose
-            self.object: Flow = flow
+                yaml_source = yaml_file.read()
+
+            self.object: Flow = Flow.load_config(yaml_source).build()
+            self.object.workspace_id = jinad_args.workspace_id
+
+            # In dependencies, we set `runs_in_docker` for the `gateway` and for `CompoundPod` we need
+            # `runs_in_docker` to be False. Since `Flow` args are sent to all Pods, `runs_in_docker` gets set
+            # for the `CompoundPod`, which blocks the requests. Below we unset that (hacky & ugly)
+            for pod in self.object._pod_nodes.values():
+                runtime_cls = update_runtime_cls(pod.args, copy=True).runtime_cls
+                if runtime_cls in ['ZEDRuntime'] + list(GATEWAY_RUNTIME_DICT.values()):
```

Comment:
Lets add a comment here explaining that particular list of runtime classes? E.g. GrpcDataRuntime is not here, which I think is correct as it should not be a CompoundPod.